### PR TITLE
ENH: Bump to ITK v5.3rc04

### PR DESCRIPTION
**Pull request by tbirdso**
Bump to most recent ITK version. May require bumping dependencies first.

### CODE REVIEW

File: .github/workflows/build-test-package.yml [hunk 1]
```diff
@@ -142,7 +142,7 @@ jobs:
       matrix:
         python-version: [37, 38, 39, 310]
         include:
-          - itk-python-git-tag: "v5.3rc03"
+          - itk-python-git-tag: "v5.3rc04"
```
Comment (dzenanz) [inline .github/workflows/build-test-package.yml:145]: This tag does not exist yet. This probably results in it being ignored.

**Review by dzenanz — COMMENTED** [review]
I understand impatience, but I advise not to spend time trying to update CIs prematurely.

File: .github/workflows/build-test-package.yml (hunk 1 above)
Comment (tbirdso) [reply .github/workflows/build-test-package.yml:145]: > This tag does not exist yet. This probably results in it being ignored.

You are right that the C++ repo is not tagged, but `itk-python-git-tag` is used to reference the ITKPythonPackage wheel archive to download, which has been tagged for ITK v5.3rc04. We should be able to use this wheel for building external module wheels as well. In the meantime for C++ builds I have supplied the ITK commit tag used in the [ITKPythonPackage v5.3rc04 builds](https://github.com/InsightSoftwareConsortium/ITKPythonPackage/blob/de54e148584827cc3cb89898358510877762f1b7/CMakeLists.txt#L97).

[export itk-python-git-tag for ITKPythonPackage scripts](https://github.com/KitwareMedical/ITKUltrasound/blob/master/.github/workflows/build-test-package.yml#L171)

[use ITK_PACKAGE_VERSION to download corresponding wheel archive](https://github.com/InsightSoftwareConsortium/ITKPythonPackage/blob/master/scripts/dockcross-manylinux-download-cache-and-build-module-wheels.sh#L50)

[ITKPythonPackage v5.3rc04 wheel artifacts](https://github.com/InsightSoftwareConsortium/ITKPythonPackage/releases/tag/v5.3rc04)

With ITKPythonPackage wheels available we _should_ be able to build corresponding wheels for external modules.

**Comment by tbirdso** [comment]
@dzenanz 

> I understand impatience, but I advise not to spend time trying to update CIs prematurely.

My apologies for not clearly stating the PR intention. @thewtex and I spoke after the hackathon today on next steps for external modules with v5.3rc04, my understanding of the current goal is to ensure that external modules build with the ITK v5.3rc04 wheels before C++ release notes are published.

My initial attempt to upgrade ITKVkFFTBackend [failed](https://github.com/InsightSoftwareConsortium/ITKVkFFTBackend/pull/36), apparently due to a pathing issue. My first thought was to attempt a similar bump on another external module to check whether this is an issue with ITKPythonPackage or with the ITKVkFFTBackend module. 

I now realize that ITKUltrasound is not the best module for comparison because it depends on several other external modules. I would like to focus efforts on bumping [ITKSplitComponents](https://github.com/InsightSoftwareConsortium/ITKSplitComponents/pull/51) to build and package with v5.3rc04 wheels because its build structure is much less complicated.

Roadmap:
- Investigate Python CI failures in ITKSplitComponents with v5.3rc04
- Possibly make changes to ITKPythonPackage / rebuild wheels if necessary
- Bump ITKSplitComponents and other external modules (ITKVkFFTBackend, maybe ITKUltrasound and its remaining dependencies)
- Publish release notes for ITK v5.3rc04

**Comment by thewtex** [comment]
To simplify version configuration and make it easier to use other remote modules, in the process of makes these updates to ITKUltrasound, related modules, we should adopt the approach @aylward has [taken in ITKTubeTK's configuration](https://github.com/InsightSoftwareConsortium/ITKTubeTK/blob/7d643c7fc6e7e6d08b3a1a1eba5696c8718adb64/.github/workflows/build-test-package.yml#L5-L9) in these modules, then add that to the ITKModuleTemplate.

**Comment by tbirdso** [comment]
ITK v5.3rc04 build archives are now stable (see [https://github.com/insightsoftwareconsortium/itksplitcomponents](https://github.com/insightsoftwareconsortium/itksplitcomponents) for example of successful v5.3rc04 package). Pushing this forward as a prerequisite for [https://github.com/KitwareMedical/ITKUltrasound/pull/182](https://github.com/KitwareMedical/ITKUltrasound/pull/182).

**Comment by tbirdso** [comment]
Note linter failures are tracked in [https://github.com/InsightSoftwareConsortium/ITKSplitComponents/issues/52](https://github.com/InsightSoftwareConsortium/ITKSplitComponents/issues/52) and result from a recent git security patch. A fix appears to be forthcoming from Github Actions. For now linter failures should not be considered blocking.

**Comment by tbirdso** [comment]
@dzenanz @thewtex Ping for review

**Comment by tbirdso** [comment]
Note linter issue has been resolved by Github Actions

**Review by dzenanz — APPROVED** [review]

**Pull request merged**